### PR TITLE
iOS manifest for iOS only

### PR DESCRIPTION
**Pull request by gerard33**
Only load the iOS manifest on iOS devices.
Also automatic reload when manifest is changed.

### CODE REVIEW

**Comment by gerard33** [comment]
@t0mg I hope this works.
I added a javascript code to only load the ios.manifest on iOS devices.
Also some code from https://www.html5rocks.com/en/tutorials/appcache/beginner/ to automatically reload the page if the ios.manifest is changed.

Seems to work here with iOS and Chrome on Windows 7 (needed to clear to AppCache on Chrome to get rid of the ios.manifest, but after that no more ios.manifest on Chrome).

Let me know if it works for you and what you think of it :)

File: public/index.html [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <!doctype html>
-<html manifest="%PUBLIC_URL%/ios.manifest" lang="en">
+<script type="text/javascript">
+if(navigator.userAgent.match(/(iPhone|iPod|iPad)/i)) {
+  document.write('<html manifest="%PUBLIC_URL%/ios.manifest" lang="en">');
+} else {
+  document.write('<html lang="en">');
+}
+</script>
```
Comment (t0mg) [inline public/index.html]: Removing the <html> tag here makes the whole document invalid. Also, html element can only contain a head and a body tag.

Can you try this?
```
<!doctype html>
<html lang="en">
<head>
  all the meta tags...
  <script>
    if (navigator.userAgent.match(/(iPhone|iPod|iPad)/i)) {
      // BTW, this test is weak, eg IE Mobile also mentions iPhone in its user agent string...
      const htmlEl = document.getElementsByTagName('html')[0];
      htmlEl.setAttribute('manifest', '%PUBLIC_URL%/ios.manifest');

      // The rest of your manifest update detection code goes here, inside the user agent test.
      // ...
    }
  </script>
</head>
<body>
...
```

**Review by t0mg — CHANGES_REQUESTED** [review]
We need to keep index.html a valid html file. See my other comment for something we can try. If it doesn't work (ie fails at loading the manifest on iOS), and there is absolutely no other way for iOS to work, then I'd rather keep the manifest for everyone.

Thanks!

File: public/index.html (hunk 1 above)
Comment (gerard33) [reply public/index.html]: Tried this and it works on iOS.

**Review by t0mg — COMMENTED** [review]
Thanks for the update! So does it still works on iOS this way?

**Comment by gerard33** [comment]
Yeah, if ios.manifest is changed this is noticed on iOS.

**Pull request merged**